### PR TITLE
feat: Added `aiplatform.Model.update`  method

### PR DESCRIPTION
**Pull request by ivanmkc**
Added ability to update `display_name`, `description` and `labels`.

Fixes https://b.corp.google.com/issues/213921337

# TODO
- [x] Manual test
- [x] Automated unit test
- [x] Automated integration test

# Test code
```

models = Model.list()

my_model = models[0]

print(f"eTag: {my_model._gca_resource.etag}")
print(f"Current display_name: {my_model.display_name}")
my_model = my_model.update(display_name=my_model.display_name + "_new")
print(f"Current display_name: {my_model.display_name}")

print(f"eTag: {my_model._gca_resource.etag}")
my_model = Model(my_model.resource_name)  # Fetch model from server
print(f"eTag after fetch: {my_model._gca_resource.etag}")
print(f"Current description: {my_model.description}")
my_model = my_model.update(description=my_model.description + "_new")
print(f"Current description: {my_model.description}")

print(f"eTag: {my_model._gca_resource.etag}")
my_model = Model(my_model.resource_name)  # Fetch model from server
print(f"eTag after fetch: {my_model._gca_resource.etag}")
print(f"Current labels: {my_model.labels}")
my_model = my_model.update(labels={"my_label": "updated"})
print(f"Current labels: {my_model.labels}")
```

# Bug?
There's some weirdness going on with the eTags though. If I don't "refetch" the model from the server, I get an "etag mismatch" exception on the subsequent update.

Tracked at b/214432914

### CODE REVIEW

File: google/cloud/aiplatform/models.py [hunk 1]
```diff
@@ -1502,6 +1501,90 @@ def __init__(
         )
         self._gca_resource = self._get_gca_resource(resource_name=model_name)
 
+    @base.optional_sync()
+    def update(
+        cls,
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/models.py:1506]: This should be an instance method and should update the instance.

**Review by sasha-gitg — CHANGES_REQUESTED** [review]
Please add or update an integration test to use this method:

An additional test case can be added here using the same model: https://github.com/googleapis/python-aiplatform/blob/main/tests/system/aiplatform/test_model_upload.py

It can also be added to this e2e test: https://github.com/googleapis/python-aiplatform/blob/main/tests/system/aiplatform/test_e2e_tabular.py

File: google/cloud/aiplatform/models.py [hunk 2]
```diff
@@ -1502,6 +1501,73 @@ def __init__(
...
+
+        update_mask = field_mask_pb2.FieldMask(paths=update_mask)
+
+        model = self.api_client.update_model(
+            model=current_model_proto, update_mask=update_mask
+        )
+
+        self._gca_resource = model
+
+        return self
```
[the remark this answers is not in the document]
Comment (ivanmkc) [reply google/cloud/aiplatform/models.py:1571]: Done

File: google/cloud/aiplatform/models.py [hunk 3]
```diff
@@ -1502,6 +1501,73 @@ def __init__(
...
+                are allowed.
+                See https://goo.gl/xmQnxf for more information
+                and examples of labels.
+        Returns:
+            model: Updated model resource.
+        Raises:
+            ValueError: If `labels` is not the correct format.
+        """
+
+        current_model_proto = self.gca_resource
```
[the remark this answers is not in the document]
Comment (ivanmkc) [reply google/cloud/aiplatform/models.py:1542]: Done.

File: google/cloud/aiplatform/models.py [hunk 4]
```diff
@@ -1502,6 +1501,73 @@ def __init__(
         )
         self._gca_resource = self._get_gca_resource(resource_name=model_name)
 
+    def update(
+        self,
+        display_name: Optional[str] = None,
+        description: Optional[str] = None,
+        labels: Optional[Dict[str, str]] = None,
+    ):
```
[the remark this answers is not in the document]
Comment (ivanmkc) [reply google/cloud/aiplatform/models.py:1509]: Done.

File: google/cloud/aiplatform/models.py [hunk 5]
```diff
@@ -1502,6 +1501,75 @@ def __init__(
...
+
+        if labels:
+            utils.validate_labels(labels)
+
+            copied_model_proto.labels = labels
+            update_mask.append("labels")
+
+        update_mask = field_mask_pb2.FieldMask(paths=update_mask)
+
+        _ = self.api_client.update_model(
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/models.py:1565]:
```suggestion
        self.api_client.update_model(
```

**Review by sasha-gitg — APPROVED** [review]
LGTM!

**Pull request merged**